services/templates/service.go

```diff
@@ -141,3 +144,87 @@ func (s *service) DeleteTemplate(c context.Context, templateID, userID int64) er
 
 	return nil
 }
+
+// GetTemplate returns the template with given template id and user id
+func (s service) GetTemplate(c context.Context, templateID int64, userID int64) (*entities.Template, error) {
+	template, err := s.db.GetTemplate(templateID, userID)
+	if err != nil {
+		return nil, fmt.Errorf("get template: %w", err)
+	}
+
+	resp, err := s.s3.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(templatesBucket),
+		Key:    aws.String(fmt.Sprintf("%d/%d", userID, templateID)),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeNoSuchKey:
+				return nil, ErrHTMLPartNotFound
+			case s3.ErrCodeInvalidObjectState:
+				return nil, ErrInvalidHTMLPart
+			default:
+				return nil, fmt.Errorf("get object: %w", aerr)
+			}
+		}
+		return nil, fmt.Errorf("get object: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	var html []byte
+	_, err = resp.Body.Read(html)
+	if err != nil {
+		return nil, fmt.Errorf("read: %w", err)
+	}
+
+	template.HTMLPart = string(html)
+
+	return template, nil
+}
+
+/*func (s service) getTemplate(c context.Context, templateID int64, userID int64) (*entities.Template, error) {
```

Comment:
@gudgl let's test and implement this function with the errgroup. From what I can tell I think it's mostly correct.

Reply:
@FilipNikolovski i am resolving this since i commented on the previous comment about the same purpose